core/helpers/constants.ts

```diff
@@ -2,5 +2,8 @@
 
 export const HARDHAT_PRIVATE_KEY = '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80'; // deterministic private key from hardhat.
 export const NETWORK = 'custom';
-export const REMOTE_RPC_URL = process.env.REMOTE_RPC_URL || 'http://localhost:8545';
+if (!process.env.REMOTE_RPC_URL) {
+    throw new Error('Missing the required environemnt varialbe');
+}
```

Comment:
I assume this would also fail while running plain frontend or bot. I don't think that is desirable. We only need to check this during tests/at the time it is used